cirq-google/cirq_google/experimental/noise_models/fidelity.py

```diff
@@ -0,0 +1,124 @@
+import cirq
+import numpy as np
+from typing import Optional, Sequence
+from cirq.devices.noise_model import validate_all_measurements
+from itertools import product
+
+class Fidelity():
+  def __init__(
+      self,
+      *,
+      t1: Optional = None,
+      decay_constant: Optional = None,
+      xeb_fidelity: Optional = None,
+      pauli_error: Optional = None,
+      p00: Optional = None,
+      p11: Optional = None
+  ) -> None:
+    """
+    Detailed comments about arguments here
+    """
+    if not any([t1, decay_constant, xeb_fidelity, pauli_error, p00, p00]):
+      raise ValueError('At least one metric must be specified')
+
+    self._t1 = t1
+    self._p = decay_constant
+    self._p00 = p00
+    self._p11 = p11
+    self._pauli_error = pauli_error
+    self._xeb = xeb_fidelity
+
+  @property
+  def decay_constant(self):
+    return self._p
+
+  @property
+  def p00(self):
+    return self._p00
+
+  @property
+  def p11(self):
+    return self._p11
+
+  @property
+  def pauli_error(self):
+    return self._pauli_error
+
+  @property
+  def t1(self):
+    return self._t1
+
+  def decay_constant_to_xeb_error(self, decay_constant: float, N: int = 4):
+    return (1 - decay_constant) * (1 - 1 / N)
+
+  def decay_constant_to_pauli_error(self, decay_constant: float, N: int = 2):
+    return (1 - decay_constant) * (1 - 1 / N / N)
+
+  def pauli_error_to_xeb_error(self, pauli_error: float, N: int = 4):
+    decay_constant = 1 - (pauli_error / (1 - 1 / N))
+    return decay_constant_to_xeb_error(decay_constant)
+
+  def pauli_error_to_decay_constant(self, pauli_error: float, N: int = 2):
+    return 1 - (pauli_error / (1 - 1 / N / N))
+
+  def pauli_error_from_t1(self, t: float, t1: float):
+    t2 = 2 * t1
+    return (1 - np.exp(-t / t2)) / 2 + (1 - np.exp(-t / t1)) / 4
+
+  def pauli_error_from_depolarization(self, t : float):
+    if self._t1 is not None:
+      return self._pauli_error - self.pauli_error_from_t1(t, self._t1)
+    else:
+      return self._pauli_error
+
+
+_sq_inds = np.arange(4)
```

Comment:
Does this variable need to be at this module level scope or could it be brought down to the `NoiseModelFromFidelity` class ?